crates/dslab-iaas/src/core/config/exp_config.rs

```diff
@@ -0,0 +1,378 @@
+    pub simulation_length: Option<NumericParam<f64>>,
+    /// duration beetween user access the simulation info
+    pub step_duration: Option<NumericParam<f64>>,
+    /// VM becomes failed after this timeout is reached
+    pub vm_allocation_timeout: Option<NumericParam<f64>>,
+    /// Dataset of virtual machines
+    pub trace: Option<StringParam>,
+    /// cloud physical hosts
+    pub hosts: Option<Vec<HostConfig>>,
+    /// cloud schedulers
+    pub schedulers: Option<Vec<SchedulerConfigRaw>>,
+}
+
+/// Represents simulation configuration.
+#[derive(Debug, Clone)]
+pub struct ConfigState {
+    /// periodically send statistics from host to monitoring
+    pub send_stats_period: Rc<RefCell<DynamicNumericVariable<f64>>>,
+    /// message trip time from any host to any direction
+    pub message_delay: Rc<RefCell<DynamicNumericVariable<f64>>>,
+    /// when allocation request fails then wait for this duration
+    pub allocation_retry_period: Rc<RefCell<DynamicNumericVariable<f64>>>,
+    /// vm initialization duration
+    pub vm_start_duration: Rc<RefCell<DynamicNumericVariable<f64>>>,
+    /// vm deallocation duration
+    pub vm_stop_duration: Rc<RefCell<DynamicNumericVariable<f64>>>,
+    /// pack VM by real resource consumption, not SLA
+    pub allow_vm_overcommit: bool,
+    /// currently used to define VM migration duration
+    pub network_throughput: Rc<RefCell<DynamicNumericVariable<u64>>>,
+    /// length of simulation (for public datasets only)
+    pub simulation_length: Rc<RefCell<DynamicNumericVariable<f64>>>,
+    /// duration beetween user access the simulation info
+    pub step_duration: Rc<RefCell<DynamicNumericVariable<f64>>>,
+    /// VM becomes failed after this timeout is reached
+    pub vm_allocation_timeout: Rc<RefCell<DynamicNumericVariable<f64>>>,
+    /// Dataset of virtual machines
+    pub trace: Option<Rc<RefCell<DynamicDatasetVariable>>>,
+    /// cloud physical hosts
+    pub hosts: Vec<HostConfig>,
+    /// cloud schedulers
+    pub schedulers: Vec<SchedulerConfigState>,
+}
+
+/// Represents simulation configuration.
+#[derive(Clone)]
+pub struct ExperimentConfig {
+    /// config value
+    pub current_state: ConfigState,
+    /// dynamic variables which will result in multiple test cases
+    pub dynamic_variables: Vec<Rc<RefCell<dyn DynamicVariable>>>,
+    /// if the next state is first
+    pub initial_state: bool,
+    /// if there is one more state to process
+    pub has_next: bool,
+}
+
+impl ExperimentConfig {
+    /// Creates simulation config by reading parameter values from .yaml file (uses default values if some parameters are absent).
+    pub fn from_file(file_name: &str) -> Self {
+        let current_state_raw: ConfigDataRaw = serde_yaml::from_str(
+            &std::fs::read_to_string(file_name).unwrap_or_else(|_| panic!("Can't read file {}", file_name)),
+        )
+        .unwrap_or_else(|_| panic!("Can't parse YAML from file {}", file_name));
```

Comment:
```suggestion
        .unwrap_or_else(|err| panic!("Can't parse YAML from file {}: {}", file_name, err));
```
супер полезно если что-то не так в файле